src/parser/mod.rs

```diff
@@ -4055,6 +4070,44 @@ impl<'a> Parser<'a> {
         )
     }
 
+    /// Look backwards in the token stream and expect that there was only whitespace tokens until the previous newline
+    pub fn expect_previously_only_whitespace_until_newline(&mut self) -> Result<(), ParserError> {
```

Comment:
hmm so i think it would be good to write this function in a reusable manner if possible since it feels similar to the `peek_*`, `prev_*` helper functions that we have. Would it be possible to introduce a generic `prev_nth_token_no_skip(n: usize)` that can be called by parser and dialect methods?

Reply:
Done 👍